torch/_dynamo/decorators.py

```diff
@@ -206,18 +206,18 @@ def mark_dynamic(t, index):
     if is_traceable_wrapper_subclass(t):
         # default behavior: mirror mark_dynamic() on all inner tensors with same dim as t
         # TODO: Make this configurable via a supported public API
-        _apply_func_to_inner_tensors_of_same_dim(mark_dynamic, t, index)
+        _apply_func_to_inner_tensors_of_same_dim(mark_dynamic, t, index, min, max)
 
     if isinstance(index, int):
         if not hasattr(t, "_dynamo_dynamic_indices"):
             t._dynamo_dynamic_indices = set()
         # TODO(voz): Should we bounds check?
-        t._dynamo_dynamic_indices.add(index)
+        t._dynamo_dynamic_indices.add((index, min, max))
```

Comment:
Consider just making a little data structure for this, easier than indexing it

Reply:
Sure. I added `_DimRange` dataclass now to access it via `dim_range.min` etc. Please let me know if that aligns with your comment.